### DISCUSSION TITLE
Trying to send a photo+avi right after wifi connects

### DISCUSSION THREAD

**Discussion by FBMinis** [jameszah/ESP32-CAM-Video-Telegram#33] [General]
I put together a simple circuit that turns ON and keeps the camera ON while the output of the PIR is HIGH. 

![keep camera ON while PIR output is HIGH](https://user-images.githubusercontent.com/36670323/202021365-29515a9e-fbd7-4709-accc-ca5dd395c2f3.png)

I would like to receive a photo and an avi right after the camera connects to wifi (im using v8.9 to keep it simple without wifimanager)

I though I just had to call `xTaskCreatePinnedToCore( the_camera_loop, "the_camera_loop", 10000, NULL, 1, &the_camera_loop_task, 1);` by the end of setup() function but Im only receiving a photo, not the video.

The Serial Monitor seems to show the avi was recorded, but I dont received it.

```
0:37:23.218 -> ---------------------------------
20:37:23.218 -> ESP32-CAM Video-Telegram pir-cam 8.9
20:37:23.218 -> ---------------------------------
20:37:23.639 -> Pic, len=17580, new fb 3FAFF098
20:37:23.686 -> Pic, len=16799, new fb 3FB0E0A8
20:37:23.733 -> Pic, len=16650, new fb 3FB1D0B8
20:37:23.780 -> Pic, len=27505, new fb 3FB2C0C8
20:37:23.827 -> Pic, len=26104, new fb 3FAFF098
20:37:23.869 -> Pic, len=26203, new fb 3FB0E0A8
20:37:23.915 -> Pic, len=26126, new fb 3FB1D0B8
20:37:24.007 -> Connecting Wifi: Reis
20:37:24.101 -> .....mDNS responder started 'deskpir'
20:37:26.629 -> 
20:37:26.629 -> WiFi connected
20:37:26.629 -> IP address: 192.168.1.98
20:37:26.629 -> Setup PIRpin = 0, 0, 0, 0, 0,  
20:37:28.223 -> 
20:37:28.223 -> Start the avi ... at 5889
20:37:28.223 -> Framesize 8, quality 10, length 0 seconds
20:37:28.223 -> 
20:37:28.223 -> Starting an avi 
20:37:34.082 -> End the Avi
20:37:34.129 -> End of avi - closing the files
20:37:34.129 -> 
20:37:34.129 -> *** Video recorded and saved ***
20:37:34.129 -> 
20:37:34.129 -> Recorded   150 frames in     5 seconds
20:37:34.129 -> File size is 2766100 bytes
20:37:34.129 -> Adjusted FPS is 12.70
20:37:34.129 -> Max data rate is 239572 bytes/s
20:37:34.129 -> Frame duration is 78760 us
20:37:34.129 -> Average frame length is 18426 bytes
20:37:34.129 -> Writng the index, 150 frames
20:37:34.129 -> ---
20:37:34.129 -> End the avi at 11811.  It was 150 frames, 5922 ms at 30.00 fps...
20:37:34.129 -> Deleting the camera task
20:37:34.598 -> 
20:37:34.598 -> >>>>> Sending as 512 byte blocks, with jzdelay of 0, bytes=  25731
20:37:35.910 -> **********
20:37:36.708 -> 
20:37:36.708 -> 
20:37:36.708 -> Sending clip with caption
20:37:36.708 -> 
20:37:36.708 -> >>>>> Sending as 512 byte blocks, with a caption, and with jzdelay of 0, bytes=  2768144
20:37:36.755 -> done!
20:39:09.890 -> 
20:39:09.890 -> Brownot
```

**Comment by FBMinis** [comment]
SOLVED by adding the following line just before the end of the Setup function:

`bot.longPoll = 0; xTaskCreatePinnedToCore( the_camera_loop, "the_camera_loop", 10000, NULL, 1, &the_camera_loop_task, 1);
`

The PIR detects movement, pulls the gate of the n-mosfet High, which pulls the gate of the p-mosfet Low, powering the camera for as long as Tx potentiometer sets it to. A photo and an video are immediately sent upon boot, then it waits for new messages as normal.

This is suitable for battery use, like keeping it in my car outside, over my doorstep or terrace, where I don't have access to mains line.

I use 4xAA batteries because I'm eerie of lipos. They are 2200mAh, the quiescent current of the PIR is 50uA and the project consumes 170mAh while running. In theory, if it detects movement once per hour, it should run for nearly 2 months.

Any suggestions for improvements would be welcome.

**Comment by jameszah** [comment]
Interesting -- I'll have a look at that - some mods coming: add the modified radu light sensor, that hd issue, and the old picture issue

**Comment by FBMinis** [comment]
I've been able to make the project work at 3.3V consistently by using an HT7833, through which the PIR and Camera are powered:

https://www.holtek.com.tw/documents/10179/82844a36-1633-498c-a6e3-4d99f2d0f6d7

The PIR module was modified to work at 3.3V by moving the diode to the side, then connecting VCC pin directly to another pin close to the corner of the pcb where re-triggering is set.
Battery: 4xAA alkaline
Current when there's no movement: 70uA
Current while the camera is sending photo and clip: 200mA

In theory, with 10 wake-ups a day and 2000mAh batteries, it should last over a month and a half.
Battery comparison: https://lygte-info.dk/review/batteries2012/CommonAAIndividualTest%20UK.html

![IMG_20230309_125437](https://user-images.githubusercontent.com/36670323/224067374-504277a3-3728-42c2-b05e-4ff027b1e2a0.jpg)
![IMG_20230309_125545](https://user-images.githubusercontent.com/36670323/224067417-43dc628e-5798-437c-ac55-ab3c12c1aee4.jpg)
![IMG_20230309_125632](https://user-images.githubusercontent.com/36670323/224067509-ec9f8120-b56a-42bd-b296-129599d0447a.jpg)

**Reply by jameszah** [reply]
Have you tried these:
![Screenshot_20230309_091459_Chrome](https://user-images.githubusercontent.com/36938190/224085589-e2533dcf-7ec7-4032-a158-743ae352a1fc.jpg)

The big pir (with the 4 big capacitors at the corners) can power a small circuit (leds etc), but these little ones just give you a signal that a transistor or gpio can read, so less power required.  Big pir is more configurable though.

**Reply by jameszah** [reply]
Is an "event" here a single picture, or a 3 mb video?

**Reply by FBMinis** [reply]
I haven't tried other PIR modules as I already had about 10 of the 4xCapacitor ones - that's the only reason. That SR602 is smaller but seems to be well documented and has the same functionalities, I'll check them out. 

In the battery operated project, a picture, video and message are sent when movement is detected. It takes about 1min for the picture and video to be sent, the PIR can be regulated to stay ON for that amount of time or up to 7min, giving us time to request more photos or videos.

https://github.com/FBMinis/esp32camPhotoVideoUponBoot

![Screenshot_2023-03-09-20-00-33-999_org telegram messenger web](https://user-images.githubusercontent.com/36670323/224144191-a0801901-f1d3-45a9-bcc2-b69c1a699746.jpg)

**Reply by FBMinis** [reply]
[Digital Intelligent PIR Sensor BS-612.pdf](https://github.com/jameszah/ESP32-CAM-Video-Telegram/files/15108245/Digital.Intelligent.PIR.Sensor.BS-612.pdf)
![mh-sr602 zt-bm622](https://github.com/jameszah/ESP32-CAM-Video-Telegram/assets/36670323/06ca4dcf-fb13-49fd-af2d-e2cfe8ff6fcf)
